jib-core/src/test/java/com/google/cloud/tools/jib/http/TestWebServer.java

```diff
@@ -47,14 +49,26 @@
   private final Semaphore threadStarted = new Semaphore(0);
   private final StringBuilder inputRead = new StringBuilder();
 
+  private final List<String> responses;
+
   public TestWebServer(boolean https)
       throws IOException, InterruptedException, GeneralSecurityException, URISyntaxException {
+    this(https, Arrays.asList("HTTP/1.1 200 OK\nContent-Length:12\n\nHello World!"));
+  }
+
+  public TestWebServer(boolean https, List<String> responses)
+      throws IOException, InterruptedException, GeneralSecurityException, URISyntaxException {
     this.https = https;
-    serverSocket = createServerSocket(https);
-    ignoreReturn(executorService.submit(this::serve200));
+    this.responses = responses;
+    serverSocket = https ? createHttpsServerSocket() : new ServerSocket(0);
+    ignoreReturn(executorService.submit(this::serveResponses));
```

Comment:
I don't understand what this is for?

Reply:
Nothing really changed. I just renamed the method `serve200` to `serveResponses`.

Nothing really changed about assigning `serverSocket` as well.

If you are asking what this PR is for, I'm just adding one more test to verify that our `Connection` class is supporting `http(s).proxyUser/Password` properly. Although I need this test in a later PR that has not been filed, I wanted to merge this test now, since #1808 will most like require modifying `TestWebServer`, which I am modifying here to add the test.

Reply:
I meant `ignoreReturn` in general. I know it's not new, I just don't know why it's needed in the first place.

Reply:
`submit()` returns a `Future`, and it causes a warning that I am not making any use of the return value. I will update the comment to eliminate confusion.

Reply:
Oh I see